test/integration/helpers/html-utils.ts

```diff
@@ -161,9 +166,23 @@ export class HtmlElementLookup {
     if (!label) {
       return;
     }
-    const labelLastDivElement = document.querySelector<HTMLElement>(this.bpmnQuerySelectors.labelLastDiv(bpmnId));
+
+    // TODO make the label check pass with jest v28 and the previous implementation of the test
+    const labelSvgGroup = this.querySelector<HTMLElement>(this.bpmnQuerySelectors.labelSvgGroup(bpmnId));
+    const foreignObject = labelSvgGroup.querySelector('g > foreignObject');
+    // also work
+    // svg > g > g > g[data-bpmn-id="serviceTask_1_2"].bpmn-label > g > foreignObject
+    // const foreignObject = this.querySelector(`svg > g > g > g[data-bpmn-id="${bpmnId}"].bpmn-label > g > foreignObject`);
+
+    // svg > g > g > g[data-bpmn-id="serviceTask_1_2"].bpmn-label > g > foreignObject > div > div > div
+    // const labelLastDivElement = foreignObject.firstElementChild.firstElementChild.firstElementChild;
+    const labelLastDivElement = foreignObject.querySelector('div > div > div');
+
+    // Do not work anymore with jest 28 (jsdom bump for 16.6 to 19), this is due to the part of the selector after foreignObject
+    // It works well in BpmnPageSvgTester (check in the browser, not with jsdom)
+    // const labelLastDivElement = this.querySelector<HTMLElement>(this.bpmnQuerySelectors.labelLastDiv(bpmnId));
     expect(labelLastDivElement.innerHTML).toEqual(label);
-    const labelSvgGroup = document.querySelector<HTMLElement>(this.bpmnQuerySelectors.labelSvgGroup(bpmnId));
+    // const labelSvgGroup = this.querySelector<HTMLElement>(this.bpmnQuerySelectors.labelSvgGroup(bpmnId));
```

Comment:
@csouchet I have not been able to make this test pass with the previous test implementation. I spend some time on this topic and I am stuck.
I have been able to make it pass by changing the test implementation but this introduces a duplication with e2e tests and expose the selectors which is not what we want.

Can you help me solve this problem?

Reply:
ℹ️ I have just tested with jest@29 and the same problem occurs.

Reply:
I found that `Jest 28` has upgraded from `jsdom@16` to `jsdom@19`: https://jestjs.io/blog/2022/04/25/jest-28#jsdom-19
Maybe the problem come from here.

Reply:
❓ Also, maybe we should use `jest-environment-node` for the integration test and not `jest-environment-jsdom`: https://jestjs.io/blog/2022/04/25/jest-28#packagejson-exports

Reply:
No we cannot.
We use `jest-environment-jsdom` in some unit tests (config only in tests that require it, declaration with JSdoc comment block) and in all integration tests (via jest configuration) because the code under test requires access to DOM element that are not available with `jest-environment-node` which is the default Jest environment.

Reply:
I found a workaround for this problem in 8256c59d